go/ics23.go

```diff
@@ -36,11 +36,19 @@ type CommitmentRoot []byte
 func VerifyMembership(spec *ProofSpec, root CommitmentRoot, proof *CommitmentProof, key []byte, value []byte) bool {
 	// decompress it before running code (no-op if not compressed)
 	proof = Decompress(proof)
-	ep := getExistProofForKey(proof, key)
+	comparedKey, err := DoHashOrNoop(spec.GetPrehashComparedKey(), key)
+	if err != nil {
```

Comment:
What about renaming `key` to `preimageKey` and then renaming `comparedKey` to `key` or `keyInTree`? 

From my understanding, the key passed in, is a preimage of the key used in the actual tree. This addition will then allow the tree to hash the preimage if it so chooses. I find it easier to reason about this way. Thoughts?